src/main/java/org/opensearch/neuralsearch/query/visitor/NeuralSearchQueryVisitor.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.query.visitor;
+
+import java.util.Map;
+
+import org.apache.lucene.search.BooleanClause;
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.index.query.QueryBuilderVisitor;
+import org.opensearch.neuralsearch.query.NeuralQueryBuilder;
+
+public class NeuralSearchQueryVisitor implements QueryBuilderVisitor {
+
+    private String modelId;
+    private Map<String, Object> neuralFieldMap;
+
+    public NeuralSearchQueryVisitor(String modelId, Map<String, Object> neuralFieldMap) {
+        this.modelId = modelId;
+        this.neuralFieldMap = neuralFieldMap;
+    }
+
+    @Override
+    public void accept(QueryBuilder queryBuilder) {
```

Comment:
as per my understanding the logic here should be :
1. If model id is provided in query then do nothing.
2. If model id is present in the field map, use that model id
3. If no field is not in the field map, then use the model id provided in the SearchRequestProcessor. 

The current code doesn't seem to work like that. Can you share some light on the logic.

Reply:
cool, lets fix the logic.